dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/appmanager/JCloudsAppStorageService.java

```diff
@@ -431,7 +442,10 @@ public void deleteApp(App app) {
       blobStore.removeBlob(config.container, resource.getName());
     }
 
-    reservedNamespaces.remove(app.getActivities().getDhis().getNamespace(), app);
+    String namespace = app.getActivities().getDhis().getNamespace();
+    if (namespace != null) {
```

Comment:
FYI: Not sure why nobody ran into this NPE before because most apps do not have a namespace and this then would cause a NPE when trying to remove a `null` key from the map. Maybe this just "worked" because this was the last step so even if it failed with a NPE - at that point the deletion itself was already performed so the effect was still there just with an error message.

Reply:
The use of HashMap allows nulls so this was handled by the library I think. It never would've propagated. I see it's been updated to use ConcurrentHashMap which would throw a NPE.